test/unit/host_test.rb

```diff
@@ -164,6 +164,20 @@ class HostTest < ActiveSupport::TestCase
     assert_equal host.vm_compute_attributes, nil
   end
 
+  test "can view hosts as non-admin user" do
+    #Testing Issue #13545
+    h = FactoryGirl.create(:host, :managed)
+    user_role  = FactoryGirl.create(:user_user_role)
+    user       = user_role.owner
+    role       = user_role.role
+    permission = Permission.find_by_name('view_hosts')
+    FactoryGirl.create(:filter, :on_name_all, :role => role, :permissions => [permission])
```

Comment:
You can save yourself from all this role/filter/permission/user setup, use `setup_user('view', 'hosts')` if all you want is an user with permission to view all hosts. Then you don't need to use `User.as` in the lines below, as `User.current` is set to that user.